util/secret/certificates.go

```diff
@@ -215,17 +233,31 @@ func (c Certificates) Generate() error {
 				continue
 			case ServiceAccount:
 				generator = generateServiceAccountKeys
+			case ClusterAPIEtcdClient:
+				continue
 			default:
 				generator = generateCACert
 			}
-
-			kp, err := generator()
+			var kp *certs.KeyPair
+			var err error
+			if generator != nil {
+				kp, err = generator()
+			}
+			if certificate.Purpose == ClusterAPIEtcdClient {
```

Comment:
i think this is impossible to reach? if the `certificate.Purpose == ClusterAPIEtcdClient` it would have already hit the `continue` block in the case statement.

Reply:
Messed that up. Have rewritten to match the intent.